client/src/js/services/JournalPostingModalService.js

```diff
@@ -0,0 +1,173 @@
+angular.module('bhima.services')
+  .service('JournalPostingModalService', JournalPostingModalService);
+
+/** Dependencies injection */
+JournalPostingModalService.$inject = ['util', '$http', '$uibModal'];
+
+function JournalPostingModalService(util, $http, Modal) {
+  var service = this;
+  var baseUrl = '/trial_balance/';
+
+  service.parseSelectedGridRecord = parseSelectedGridRecord;
+  // service.postingModalService = postingModalService;
+  // service.getCurrentGroupingColumn = getCurrentGroupingColumn;
+  service.switchGroup = switchGroup;
+  service.getDataByAccount = getDataByAccount;
+  service.checkTransactions = checkTransactions;
+  service.getFeedBack = getFeedBack;
+  service.getCSSClass = getCSSClass;
+  service.openErrorViewerModal = openErrorViewerModal;
+  service.postToGeneralLedger = postToGeneralLedger;
+
+  function openErrorViewerModal(errors, feedBack, cssClass) {
+
+    return Modal.open({
+      templateUrl: 'partials/journal/modals/journalErrorViewer.modal.html',
+      controller:  'JournalErrorViewerModalController as JournalErrorViewerModalCtrl',
+      size : 'lg',
+      resolve : {
+        params : function getParams() { return {errors : errors, feedBack : feedBack, cssClass : cssClass}; }
+      }
+    });
+
+  }
+
+  function  getCSSClass (feedBack) {
+    return feedBack.hasError ? 'error' : feedBack.hasWarning ? 'warning' : 'success';
+  }
+
+  function getFeedBack(errors) {
+    var feedBack = {};
+
+    feedBack.hasError = errors.some(function (error) {
+      if(error){
+        return error.fatal;
+      }else{
+        return false;
+      } 
+    });
+    
+    if(feedBack.hasError){
```

Comment:
All errors are returned at once, but I was sending feedback based on error priority fatal error first, and than no fatal error. But I have changed it now the entire feedback is sent.